ouroboros-network/test/Test/Mux.hs

```diff
@@ -148,6 +156,51 @@ newtype DummyPayload = DummyPayload {
 instance Show DummyPayload where
     show d = printf "DummyPayload %d\n" (BL.length $ unDummyPayload d)
 
+-- |
+-- Generate a byte string of a given size.
+--
+genByteString :: Int64 -> Gen BL.ByteString
+genByteString size = MkGen $ \(QCGen g0) _ ->
+  if size <= 0
+    then BL.empty
+    else BL.unfoldr gen (size, g0)
+  where
+    gen :: (Int64, SM.SMGen) -> Maybe (Word8, (Int64, SM.SMGen))
+    gen (!i, !g)
+        | i <= 0    = Nothing
+        | otherwise = Just (fromIntegral w64, (i - 1, g'))
```

Comment:
It looks like you're throwing away 3 random bytes for every random byte that you use which seems a bit wasteful for an efficient ByteString generator ;)

Reply:
What do you mean?

Reply:
>It looks like you're throwing away 3 random bytes...
That should be 7 bytes.

Reply:
The `gen` step only generates a random `Word8` each iteration.
Instead of getting one byte of random `Word8` from `w64` you could get 8.
```
let b0 = fromIntegral $ w64
     b1 = fromIntegral $ shiftR w64 8
     b2 = fromIntegral $ shiftR w64 16
     ...
    b7 = fromIntegral $ shiftR w64 56
```
And them add b0,b1, ..., b7 to the bytestring.
The end result would be more complex, but I suspect that it could be faster.

Reply:
Also, it was not meant as a serious suggestion. Just a Tongue-in-cheek comment.

Reply:
I see, I'll change it.

Reply:
There is no way to get around it, and use `BL.unfoldr` and `SM.nextWord64` at the same time; There is no `SM.nextWord8`.  It's fast enough, so we can live with it.